src/transformers/models/encodec/modeling_encodec.py

```diff
@@ -803,14 +863,388 @@ def forward(
+        stride (`Tuple[int, int]`, *optional*, defaults to `(1, 2)`):
+            Stride for the convolutional layers.
+        dilations (`List[int]`, *optional*, defaults to `<factory>`):
+            List of dilations for the convolutional layers.
+        max_filters (`int`, *optional*, defaults to 1024):
+            Maximum number of filters in the convolutional layers.
+        filters_scale (`int`, *optional*, defaults to 2):
+            Scaling factor for the number of filters in each convolutional layer.
+        normalized (`bool`, *optional*, defaults to `True`):
+            Whether to normalize the STFT.
+        norm (`str`, *optional*, defaults to `"weight_norm"`):
+            Normalization method to use.
+        activation (`str`, *optional*, defaults to `"LeakyReLU"`):
+            Activation function to use.
+        activation_params (`Dict`, *optional*, defaults to `<factory>`):
+            Parameters for the activation function.
+        output_hidden_states (`bool`, *optional*, defaults to `False`):
+            Whether to return the hidden states of each discriminator layer.
+        output_attentions (`bool`, *optional*, defaults to `False`):
+            Whether to return the attentions tensors of all attention layers.
+        return_dict (`bool`, *optional*, defaults to `False`):
+            Whether to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        torchscript (`bool`, *optional*, defaults to `False`):
+            Whether the model is used with torchscript.
+    """
+
+    model_type: str = "encodec_discriminator"
+    filters: int = 32
+    in_channels: int = 1
+    out_channels: int = 1
+    n_ffts: list = field(default_factory=lambda: [1024, 2048, 512])
+    hop_lengths: list = field(default_factory=lambda: [256, 512, 128])
+    win_lengths: list = field(default_factory=lambda: [1024, 2048, 512])
+    kernel_size: tuple = (3, 9)
+    stride: tuple = (1, 2)
+    dilations: list = field(default_factory=lambda: [1, 2, 4])
+    max_filters: int = 1024
+    filters_scale: int = 2
+    normalized: bool = True
+    norm: str = "weight_norm"
+    activation: str = "LeakyReLU"
+    activation_params: dict = field(default_factory=lambda: {"negative_slope": 0.2})
+    output_hidden_states: bool = False
+    output_attentions: bool = False
+    return_dict: bool = False
+    torchscript: bool = False
+
+
+class ConvLayerNorm(nn.LayerNorm):
+    """
+    Convolution-friendly LayerNorm that moves channels to last dimensions
+    before running the normalization and moves them back to original position right after.
+    """
+
+    def __init__(self, normalized_shape: tp.Union[int, tp.List[int], torch.Size], **kwargs):
+        super().__init__(normalized_shape, **kwargs)
+
+    def forward(self, x):
+        # Move time dimension to second position
+        ndim = len(x.shape)
+        perm = [0] + [ndim - 1] + list(range(1, ndim - 1))
+        x = x.permute(perm)
```

Comment:
This seems quite costly to run, isn't it a better way to do it ? If `ndim` is always the same, let's hardcode it

Reply:
If it's always the same, there's no need to create a class just for it, as it makes readibility more difficult. Let's use `LayerNorm` instead and do permutation around it

Reply:
Ah right yea of course, this can be simplified a lot